pkg/cloudprovider/providers/gce/gce_loadbalancer_external.go

```diff
@@ -509,29 +547,37 @@ func (gce *GCECloud) ensureTargetPoolAndHealthCheck(tpExists, tpNeedsRecreation
 	return nil
 }
 
+func (gce *GCECloud) createHealthCheckAndFirewall(svc *v1.Service, name, serviceName, ipAddress, region, clusterID string, hosts []*gceInstance, hc *compute.HttpHealthCheck) (*compute.HttpHealthCheck, error) {
+	// Check whether it is nodes health check, which has different name from the load-balancer.
+	isNodesHealthCheck := hc.Name != name
+	if isNodesHealthCheck {
+		// Lock to prevent necessary nodes health check / firewall gets deleted.
+		gce.sharedResourceLock.Lock()
+		defer gce.sharedResourceLock.Unlock()
+	}
+
+	if err := gce.ensureHttpHealthCheckFirewall(svc, serviceName, ipAddress, region, clusterID, hosts, hc.Name, int32(hc.Port), isNodesHealthCheck); err != nil {
+		return nil, err
+	}
+	var err error
+	hcRequestPath, hcPort := hc.RequestPath, hc.Port
+	if hc, err = gce.ensureHttpHealthCheck(hc.Name, hc.RequestPath, int32(hc.Port)); err != nil || hc == nil {
+		return nil, fmt.Errorf("Failed to ensure health check for %v port %d path %v: %v", name, hcPort, hcRequestPath, err)
+	}
+	return hc, nil
+}
+
 func (gce *GCECloud) createTargetPoolAndHealthCheck(svc *v1.Service, name, serviceName, ipAddress, region, clusterID string, hosts []*gceInstance, hc *compute.HttpHealthCheck) error {
 	// health check management is coupled with targetPools to prevent leaks. A
 	// target pool is the only thing that requires a health check, so we delete
 	// associated checks on teardown, and ensure checks on setup.
 	hcLinks := []string{}
 	if hc != nil {
-		// Check whether it is nodes health check, which has different name from the load-balancer.
-		isNodesHealthCheck := hc.Name != name
-		if isNodesHealthCheck {
-			// Lock to prevent necessary nodes health check / firewall gets deleted.
-			gce.sharedResourceLock.Lock()
-			defer gce.sharedResourceLock.Unlock()
-		}
-
-		if err := gce.ensureHttpHealthCheckFirewall(svc, serviceName, ipAddress, region, clusterID, hosts, hc.Name, int32(hc.Port), isNodesHealthCheck); err != nil {
+		if hc, err := gce.createHealthCheckAndFirewall(svc, name, serviceName, ipAddress, region, clusterID, hosts, hc); err != nil {
 			return err
+		} else {
```

Comment:
Would be great to get rid of this else block indent:
```golang
		hc, err := gce.createHealthCheckAndFirewall(svc, name, serviceName, ipAddress, region, clusterID, hosts, hc)
		if err != nil {
			return err
		}
		hcLinks = append(hcLinks, hc.SelfLink)
```

Reply:
Done.